skymarshal/legacyserver/server.go

```diff
@@ -29,12 +29,18 @@ func NewLegacyServer(config *LegacyConfig) (http.Handler, error) {
 	handlers := map[string]http.Handler{
 		LoginRoute: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			u := url.URL{
-				Scheme: r.URL.Scheme,
-				Host:   r.URL.Host,
-				Path:   "/sky/login",
+				Scheme:   r.URL.Scheme,
+				Host:     r.URL.Host,
+				Path:     "/sky/login",
+				RawQuery: r.URL.RawQuery,
 			}
 
-			http.Redirect(w, r, u.String(), http.StatusMovedPermanently)
+			flyPort := r.FormValue("fly_port")
```

Comment:
@vito last week @pivotal-jwinters had some good criticism about this parameter being handled in the `legacyserver` relating to the medium-term technical roadmap for killing `skymarshal` and putting cookie- and login-related stuff into a slightly smarter web server. I think they could articulate it better than I could, though.